backend/corpora/common/entities/dataset.py

```diff
@@ -86,31 +74,21 @@ def update(
             if artifacts:
                 for af in self.artifacts:
                     self.db.delete(af)
-                new_db_objects = self._create_sub_objects(
-                    artifacts, DbDatasetArtifact, add_columns=dict(dataset_id=self.id)
-                )
-                self.db.session.add_all(new_db_objects)
-                kwargs["artifacts"] = new_db_objects
+                new_objs = [DbDatasetArtifact(dataset_id=self.id, **art) for art in artifacts]
+                self.db.session.add_all(new_objs)
+                kwargs["artifacts"] = new_objs
```

Comment:
Why are you updating `kwargs` here?

Reply:
it can be removed. I was trying to avoid making lots of extra changes